pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -162,6 +162,21 @@ func (c *Controller) clusterBackupSync(ctx context.Context, interval time.Durati
 	return nil
 }
 
+type FilterFunc = func(backup *v3.EtcdBackup) bool
+
+func filterBackups(backups []*v3.EtcdBackup, filter ...FilterFunc) []*v3.EtcdBackup {
+	ret := []*v3.EtcdBackup{}
+	for _, backup := range backups {
+		for _, f := range filter {
+			if !f(backup) {
+				continue
+			}
+		}
+		ret = append(ret, backup)
+	}
```

Comment:
It seems that this implementation will add every backup to the return slice, regardless of the output of the filter functions.

Reply:
Good catch, this actually seems like a good candidate for a unit test